assignments/arrays.js

```diff
@@ -63,32 +63,54 @@ let inventory = [{"id":1,"car_make":"Lincoln","car_model":"Navigator","car_year"
-let carYears = [];
-console.log();
-
-// ==== Challenge 5 ====
-// The car lot manager needs to find out how many cars are older than the year 2000. Using the carYears array you just created, find out how many cars were made before the year 2000 by populating the array oldCars and logging it's length.
-let oldCars = [];
-console.log(); 
-
-// ==== Challenge 6 ====
-// A buyer is interested in seeing only BMW and Audi cars within the inventory.  Return an array that only contains BMW and Audi cars.  Once you have populated the BMWAndAudi array, use JSON.stringify() to show the results of the array in the console.
-let BMWAndAudi = [];
-console.log();
+//console.log(`Car 33 is a *car year goes here* *car make goes here* *car model goes here*` );
+// for(let i = 1; i < inventory.length; i++){
+//     if(i === 33){
+//         console.log(`Car 33 us a ${inventory[i].car_year} ${inventory[i].car_make} ${inventory[i].car_model}`);
+
+//     }
+
+// }
+
+// // ==== Challenge 2 ====
+// // The dealer needs the information on the last car in their inventory.  What is the make and model of the last car in the inventory?  Log the make and model into the console.
+// let lastCar = inventory[inventory.length - 1];
+// //console.log('Make: ',lastCar.car_make, '\n',  'Model: ', lastCar.car_model);
+
+// // ==== Challenge 3 ====
+// // The marketing team wants the car models listed alphabetically on the website. Sort all the car model names into alphabetical order and log the results in the console
+// let carModels = [inventory.car_model];
+// inventory.forEach(function(car){
+//     carModels.push(car.car_model)
+// });
+// //console.log(carModels.sort());
+
+// // ==== Challenge 4 ====
+// // The accounting team needs all the years from every car on the lot. Create a new array from the dealer data containing only the car years and log the result in the console.
+// let carYears = [];
+// inventory.forEach(car => carYears.push(car.car_year));
+// //console.log(carYears);
+
+// // ==== Challenge 5 ====
+// // The car lot manager needs to find out how many cars are older than the year 2000. Using the carYears array you just created, find out how many cars were made before the year 2000 by populating the array oldCars and logging it's length.
+// let oldCars = [];
+// carYears.forEach(car => {
+//     if(car < 2000){
+//         oldCars.push(car);
+//     }
+// })
+// console.log(oldCars.length); 
+
+// // ==== Challenge 6 ====
+// // A buyer is interested in seeing only BMW and Audi cars within the inventory.  Return an array that only contains BMW and Audi cars.  Once you have populated the BMWAndAudi array, use JSON.stringify() to show the results of the array in the console.
+// let BMWAndAudi = [];
+
+// inventory.forEach(function(car){
+//     if(car.car_make === "BMW" || car.car_make === "Audi"){
+//         BMWAndAudi.push(car);
+//     }
```

Comment:
Instead of forEach() I would use filter(). ForEach is good when you want to manipulate data, i.e: Appending a new property to the car object, but you just want to filter here so .filter() should be the go to.
```
let BMWAndAudi = inventory.filter(car => {
    if (car.car_make === "BMW" || car.car_make === "Audi") return car;
})

console.log(JSON.stringify(BMWAndAudi));
```